Migrate/group set selector

returnMarcco opened the pull request:
# Description

- This PR contains changes for migration the `group-set-selector` component from Coffeescript/AngularJS to Typescript/Angular.

Fixes # (issue)
- https://planner.cloud.microsoft/webui/v1/plan/njykIFLDn0iAY1at7tACfcgADgBS/view/grid/task/d6Zm3rI4q0e4t5wvnIlrkcgAHwst?tid=d02378ec-1688-46d5-8540-1c28b5f470f6

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
     - **Important:** As this Angular component emits events using Angular syntax, the parent component's declaration of this component does not currently bind to this emitted event properly. A workaround would have been to declare this component in the parent using AngularJS syntax, but I chose to use Angular syntax as the parent component can be upgraded to Angular next with less confusion. I will leave a note regarding this issue in the Planner ticket for the parent component.
- [x] This change requires a documentation update (README migration todo list)
- [x] Frontend migration

# How Has This Been Tested?

- Logging in as an admin
- Navigation to the `Unit` settings
- Selecting a unit
- Navigating to `Groups`
- Testing `Unit Group` dropdown functionality and event handling by stepping through the code in devtools
- Confirming that the parent component is not properly handling the emitted event when selecting a group set from the dropdown. 

- Note: This dropdown may not be visible by default. To make it visible for testing purposes, find its selector declaration and change the `ngIf` value to true.

# Screenshots

Before
<img width="980" height="232" alt="group_set_selector_before" src="https://github.com/user-attachments/assets/e8de9cb4-72e5-4d9d-896a-e2baf349c6c6" />

After
<img width="766" height="783" alt="Screenshot 2025-08-03 200834" src="https://github.com/user-attachments/assets/76f5baa9-00f7-4208-9be4-e158fa6c477a" />

## Testing Checklist:

- [x] Tested in latest Chrome
- [ ] Tested in latest Safari
- [ ] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

returnMarcco committed:
chore: migrate group-set-selector (WIP)

- define and declare new component files

- import new component into app

- downgrade component

returnMarcco committed:
chore: migrate group-set-selector

- convert AngularJS template syntax to Angular syntax
- finish implementing `group-set-selector.component.ts`
- remove redundant Coffeescript file

returnMarcco committed:
chore: migrate group-set-selector

- finish defining Typescript logic
- update parent component to use Angular syntax for child component declaration

returnMarcco committed:
chore: cleanup and documentation

- remove redundant Coffeescript file
- adjust frontend migration todo list

returnMarcco committed:
chore: remove redundant file

- remove redundant Coffeescript file

returnMarcco committed:
fix: attribute and event binding

- fix attribute and event binding syntax
- remove redundant template file

returnMarcco committed:
refactor: two-way binding

- remove redundant directive attributes

- rename EventEmitter var name in child component

returnMarcco committed:
Merge branch '9.x' into migrate/group-set-selector

returnMarcco committed:
Update README.md

returnMarcco committed:
Update README.md

b0ink commented on src/app/groups/group-set-selector/group-set-selector.component.html:
```diff
@@ -0,0 +1,7 @@
+<select
+  class="form-control"
```
Remove the `form-control` bootstrap classes - we are aiming to move away from bootstrap entirely as we do these migrations.

You may also need to wrap the `select` element in a `<mat-form-field>`

b0ink commented on src/app/groups/group-set-selector/group-set-selector.component.html:
```diff
@@ -0,0 +1,7 @@
+<select
```
Use Material UI elements: https://material.angular.dev/components/select

Swap out the `<select>` element with a `<mat-select>`, and `<option>` element with a `<mat-option>`

b0ink commented on src/app/groups/group-set-selector/group-set-selector.component.html:
```diff
@@ -0,0 +1,7 @@
+  class="form-control"
+  (ngModelChange)="selectGroupSet($event)"
+  [(ngModel)]="selectedGroupSet">
+  <option *ngFor="let gs of unit.groupSets" [ngValue]="gs">{{gs.name}}</option>
```
Optionally, you can implement newer Angular control flows using:
```tsx
@for (gs of unit.groupSets; track gs){
  <mat-option [ngValue]="gs">{{gs.name}}</mat-option>
}
```
To improve readability

b0ink reviewed (changes requested):
Hi @returnMarcco 

Your migration looks good and is almost ready to merge - just a few things that needs changing in the HTML as we move away from Bootstrap classes and move over to Material UI elements :)

returnMarcco commented:
Hi @b0ink,

Thanks for the feedback. Will make the changes and push them asap.

returnMarcco committed:
refactor: replace Bootstrap components with Angular Material

- replace Bootstrap components with Angular Material components & directives

- refactor to use modern Angular structural directives
